src/app/header/reducer.ts

```diff
@@ -1,28 +1,34 @@
 import * as ActionNames from './action-names'
+import { HeaderState, StakingActions} from './types'
 
-const initialState = {
-  networkWeight: null,
-  nextRewardIn: null,
-  weight: null
+const initialState: HeaderState = {
+  walletStakingInfo: {
+    networkWeight: -1,
```

Comment:
It's more accurate to make them optional and use `undefined`. Actually, you can even use the `Partial<T>` generic type [provided by TS](https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types) for your states.